src/debug/StaticWebAppDebugProvider.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, tasks, WorkspaceFolder } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { validateSwaCliInstalled } from '../commands/cli/validateSwaCliInstalled';
+import { tryGetApiLocations } from '../commands/createStaticWebApp/tryGetApiLocations';
+import { swaTaskLabel } from '../constants';
+import { getFunctionsApi } from '../getExtensionApi';
+import { localize } from '../utils/localize';
+import { AzureFunctionsExtensionApi } from '../vscode-azurefunctions.api';
+
+export class StaticWebAppDebugProvider implements DebugConfigurationProvider {
+    public async provideDebugConfigurations(_folder?: WorkspaceFolder, _token?: CancellationToken): Promise<DebugConfiguration[]> {
+        // Don't provide any debug configurations automatically for now
+        return []
+    }
+
+    public async resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DebugConfiguration, _token?: CancellationToken): Promise<DebugConfiguration | undefined> {
+        return await callWithTelemetryAndErrorHandling('resolveSwaDebugConfiguration', async (context: IActionContext) => {
+            if (debugConfiguration.preLaunchTask === swaTaskLabel) {
```

Comment:
Mentioned in review: could make a helper API that takes a task name as label, and returns true/false if it's SWA. For implementation, can use VSCode tasks API to query for the command, if it starts with `'swa '` it's probably a SWA task.